### PR TITLE
MGMT-7704: Fix disk validation for infra-env host

### PR DESCRIPTION
**Pull request by yevgeny-shnaidman**
Upon any update of the inventory, disk eligibility is determined
by checking it size againt the OCP cluster and operators requirements
When updating inventory for infra-env host, its isze should be checked
against the smallest OCP cluster requirement (worker or master)

# Assisted Pull Request

## List all the issues related to this PR

- [ ] New Feature
- [ ] Bug fix
- [ ] Tests
- [ ] Documentation
- [ ] CI/CD 

## What environments does this code impact?

- [ ] Automation (CI, tools, etc)
- [ ] Cloud
- [ ] Operator Managed Deployments
- [x] None

## How was this code tested?

- [ ] assisted-test-infra environment
- [ ] dev-scripts environment
- [ ] Reviewer's test appreciated
- [ ] Waiting for CI to do a full test run
- [ ] Manual (Elaborate on how it was tested)
- [x] No tests needed

## Assignees

/cc @
/cc @

## Checklist

- [ ] Title and description added to both, commit and PR.
- [ ] Relevant issues have been associated (see [CONTRIBUTING] guide)
- [ ] Reviewers have been listed
- [ ] This change does not require a documentation update (docstring, `docs`, README, etc)
- [ ] Does this change include unit-tests (note that code changes require unit-tests)

## Reviewers Checklist

- [ ] Are the title and description (in both PR and commit) meaningful and clear?
- [ ] Is there a bug required (and linked) for this change?
- [ ] Should this PR be backported?

[Kubernetes community documentation]: https://github.com/kubernetes/community/blob/master/contributors/guide/pull-requests.md#commit-message-guidelines
[CONTRIBUTING]: https://github.com/openshift/assisted-service/blob/master/CONTRIBUTING.md

### CODE REVIEW

File: internal/host/host.go [hunk 1]
```diff
@@ -319,27 +319,32 @@ func (m *Manager) updateInventory(ctx context.Context, cluster *common.Cluster,
...
 
-		inventoryStr, err = common.MarshalInventory(inventory)
-		if err != nil {
-			return err
-		}
+	infraEnv, err := common.GetInfraEnvFromDB(m.db, h.InfraEnvID)
+	if err != nil {
+		log.WithError(err).Errorf("not updating inventory - failed to find infra env %s", h.InfraEnvID.String())
+		return common.NewApiError(http.StatusNotFound, err)
+	}
```
Comment (avishayt) [inline internal/host/host.go:335]: Do we need to run this code if the host is bound (if the cluster variable got set above)?

Comment (yevgeny-shnaidman) [reply internal/host/host.go:335]: the flow is that we get all the data here ( cluster, infra env), and let the DiskIsEligble to decide how to check eligibility

File: subsystem/host_v2_test.go [hunk 2]
```diff
@@ -222,3 +233,59 @@ func updateHostV2(ctx context.Context, request *installer.V2UpdateHostParams) *m
...
 }
+
+func isHostInStateV2(ctx context.Context, infraEnvID strfmt.UUID, hostID strfmt.UUID, state string) (bool, string, string) {
+	rep, err := userBMClient.Installer.V2GetHost(ctx, &installer.V2GetHostParams{InfraEnvID: infraEnvID, HostID: hostID})
+	Expect(err).NotTo(HaveOccurred())
+	h := rep.GetPayload()
+	return swag.StringValue(h.Status) == state, swag.StringValue(h.Status), swag.StringValue(h.StatusInfo)
+}
+
+func waitForHostStateV2(ctx context.Context, state string, timeout time.Duration, hosts ...*models.Host) {
```
Comment (rollandf) [inline subsystem/host_v2_test.go:244]: Maybe use `Eventually` like here: https://github.com/openshift/assisted-service/blob/master/subsystem/kubeapi_test.go#L1482

Comment (rollandf) [reply subsystem/host_v2_test.go:244]: There is also `Consistently`

Comment (yevgeny-shnaidman) [reply subsystem/host_v2_test.go]: fixed

**Pull request merged**